### PR TITLE
create mongodb encryption article

### PR DESCRIPTION
**Pull request by alex-emerich**

### CODE REVIEW

File: content/08-mongodb/14-mongodb-encryption.mdx
Comment (imchairmanm) [inline content/08-mongodb/14-mongodb-encryption.mdx:70]: These are all great links.  Pulling these out into a bulleted list is another option if you think it'd help them stand out better.

**Review by imchairmanm — APPROVED** [review]
There are some things that I think might need to be clarified, but I really like the article overall. 
 More specifically, I think the concepts, scope, and organization here is really useful.  It makes me wonder if we should make this article for our other databases just to have a baseline description of what types of security mechanisms are in place to protect data.

Let me know if you want to talk over any of the feedback I left.  There were some parts I wasn't sure about, so I'm not sure if I was too clear on my notes.

File: content/08-mongodb/14-mongodb-encryption.mdx
Comment (imchairmanm) [inline content/08-mongodb/14-mongodb-encryption.mdx:57]:
```suggestion
Encrypted data on disk remains encrypted unless an attacker has access to the encryption key(s). Encryption at rest protects sensitive data in the case of lost or stolen physical drives or an exposed database file.
```

**Review by imchairmanm — COMMENTED** [review]
This is a good improvement and I think it definitely addressed some of the issues in the original draft, but I still struggled with some of the explanations here.  Specifically, I think the symmetric vs asymmetric discussion and the conversation surrounding certificates could use some clarification.  I'm happy to help figure some of this out with you if that'd be helpful.

File: content/08-mongodb/14-mongodb-encryption.mdx
Comment (alex-emerich) [reply content/08-mongodb/14-mongodb-encryption.mdx:42]: AH! I don't know why my links are coming through from copying over. Thanks for catching.

File: content/08-mongodb/14-mongodb-encryption.mdx
Comment (alex-emerich) [reply content/08-mongodb/14-mongodb-encryption.mdx:48]: I am just going to scrap this. I have been reading about the potential for compromised public keys whereas someone would replace Bob's key with their own generated key for example. It would get discovered quickly most likely, but that could already be too late. I am just going to speak instead to the disadvantage of asymmetric encryption taking longer.

File: content/08-mongodb/14-mongodb-encryption.mdx
Comment (imchairmanm) [inline content/08-mongodb/14-mongodb-encryption.mdx:17]: Just a note that we probably want to switch this one out with something else before publication.  I don't think we've gotten much feedback on where to link to specifically yet, so your call.

File: content/08-mongodb/14-mongodb-encryption.mdx
Comment (imchairmanm) [inline content/08-mongodb/14-mongodb-encryption.mdx:48]: I'm not sure how much we need to go into this, but one comment I have is that the asymmetric keys are only used to establish the connection securely, so the asymmetric overhead is a fixed cost.  Once that is done (both parties trust each other, identity has been confirmed, whatever is outlined in the connection parameters you're using) since the asymmetric keys are used to create a symmetric key that is used for the actual data for the remainder of the connection, the overhead drops off.

So basically: the overhead of asymmetric encryption only applies once per connection (it isn't a cost you pay for every message over that connection).  As I said, I'm not sure if we need to go into that, but I wanted to clarify in case that changes how we talk about the relative cost / benefits of the asymmetric keys a bit.

File: content/08-mongodb/14-mongodb-encryption.mdx
Comment (imchairmanm) [inline content/08-mongodb/14-mongodb-encryption.mdx:55]: > This static state avoids the transmission insecurity that asymmetric encryption solves, so symmetric encryption is a strong method. 

Instead of talking about transmission insecurity, it might make sense to position this as just "you don't need asymmetric encryption for data at rest because there's never a second party involved" or whatever.  Just as a way of reiterating the idea that asymmetric encryption is a solution to privacy concerns between parties.

**Review by imchairmanm — APPROVED** [review]
Looks good! :+1: Great job with the updates.  I left a few additional comments, but they're either optional ideas or just general comments on the ideas here.

File: content/08-mongodb/14-mongodb-encryption.mdx
Comment (alex-emerich) [reply content/08-mongodb/14-mongodb-encryption.mdx:17]: Yeah agreed. I may hold off until we get a bit more definitive answer or just remove the mention of launch week. Can delay until I have the image assets as well.

**Pull request merged**
